documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_moderation_metrics.md

```diff
@@ -0,0 +1,485 @@
+
+This project should introduce no new dependencies as this builds on top of our
+existing technology stack.
+
+### Infrastructure
+
+We already use Logs Insights, CloudWatch and Django Admin in our stack and, more
+specifically, in the moderation pipeline.
+
+### Other projects or work
+
+This project depends on our implementation of the moderation workflow. By having
+this implementation plan prepared we can design moderation workflow that
+accounts from metrics and analytics from the outset and not as an afterthought.
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+For the real-time metrics, as an alternative to logging that's parsed by Logs
+Insights, we can also use an SDK like Boto to
+[put metric data](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudwatch/client/put_metric_data.html)
+or
+[put_log_events](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs/client/put_log_events.html)
+into CloudWatch. This approach was not preferred because we already have a
+logging setup that goes to AWS and precedent for building with Logs Insights.
+This approach would have caused a lot of API calls to CloudWatch to put each
+event as it occurs.
+
+For the deferred metrics, as an alternative to Django Admin, we can have an
+endpoint that presents the information as JSON or as a web page. This approach
+was not preferred because Django Admin lets us build on top of the existing
+access control framework established by the
+[Django Admin implementation plan](/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md).
+That also puts the metrics closer to where the moderation is being performed and
+enables moderators to draw inferences from them.
+
+## Blockers
+
+<!-- What hard blockers exist that prevent further work on this project? -->
+
+The implementation of moderation metrics ultimately depends on the
+implementation of the moderation pipeline first.
+
+## Rollback
+
+<!-- How do we roll back this solution in the event of failure? Are there any steps that can not easily be rolled back? -->
+
+The project is purely additive with no migrations. Bugs or other problems in the
+metrics implementation can be rolled back by deploying the known last good
+version with no adverse consequences.
+
+## Privacy
+
+<!-- How does this approach protect users' privacy? -->
+
+All metrics collected by us will be aggregates and will not contain any
+information to identify a user or moderator at the individual level. Reports are
+anonymous so there is no possibility of PII in the reports. Decisions are
+associated with the maintainer that made them, so we will be careful to not make
+any reads on `maintainer_id` for our metrics.
+
+We cannot add metrics like moderator leaderboards such as the moderator with the
+most reports moderated or the fastest response times.
```

Comment:
Nit just for clarity maybe?

```suggestion
Because of this concern for anonymity, we cannot add metrics like moderator leaderboards such as the moderator with the
most reports moderated or the fastest response times.
```

Reply:
Fixed in 1d6788a.